app/graphql/mutations/profile/associate_systems.rb

```diff
@@ -12,18 +12,13 @@ class AssociateSystems < BaseMutation
 
       def resolve(args = {})
         profile = find_profile(args[:id])
-        hosts = find_hosts(args[:system_ids])
-        profile_hosts = hosts.map do |host|
-          ProfileHost.new(profile_id: profile.id, host_id: host.id)
-        end
-        ProfileHost.import!(profile_hosts)
+        add_inventory_hosts(args[:system_ids])
+        profile.update_hosts(profile.hosts.pluck(:id) | args[:system_ids])
```

Comment:
Leveraging `profile.update_hosts` here, which does nearly the same thing as the existing code, with the added benefit that it also removes hosts that are not specified. There should be no implications for this mutation used with creating a profile.